consai_examples/consai_examples/field_observer.py

```diff
@@ -611,4 +611,126 @@ def get_ball_placement_state(self, placement_position):
 
     def get_ball_pos(self):
         ball_pos = [self._ball.pos.x, self._ball.pos.y]
-        return ball_pos
\ No newline at end of file
+        return ball_pos
+
+    def get_receiver_robots_id(self, my_robot_id, select_forward_between=1):
+        # パス可能なロボットIDのリストを返す関数
+
+        # パス可能なロボットIDを格納するリスト
+        robots_to_pass = []
+        # 計算対象にする相手ロボットIDを格納するリスト
+        target_their_robots_id = []
+        # 計算上の相手ロボットの半径（通常の倍の半径（直径）に設定）
+        robot_r = 0.4
+        # 前方にいる相手ロボットの数と比較する用の変数
+        check_count = 0
+        # ロボットの位置座標取得から実際にパスを出すまでの想定時間
+        dt = 0.5
+
+        # 各ロボットの位置と速度を取得
+        our_robots_pos = self.our_robots_pos
+        their_robots_pos = self.their_robots_pos
+        our_robots_vel = self.our_robots_vel
+        their_robots_vel = self.their_robots_vel
+
+        # エラー対策
+        # TODO: 時々Vector2形式でデータが混入するので対策が必要
+        if type(our_robots_vel[0]) is not list:
+            return []
+
+        # フィールド上にいるロボットのIDのみリストに格納する
+        our_robots_in_field = [robot_id for robot_id in range(len(our_robots_pos)) if our_robots_pos[robot_id] != "None"]
+        their_robots_in_field = [robot_id for robot_id in range(len(their_robots_pos)) if their_robots_pos[robot_id] != "None"]
+
+        # パサーよりも前にいる味方ロボットIDをリストにまとめる
+        forward_our_robots_id = [robot_id for robot_id in our_robots_in_field 
+                                if robot_id != my_robot_id and our_robots_pos[my_robot_id][0] < our_robots_pos[robot_id][0] + (abs(our_robots_vel[robot_id][0]) * dt + robot_r) * 
+                                our_robots_vel[robot_id][0] / math.sqrt(our_robots_vel[robot_id][0] ** 2 + our_robots_vel[robot_id][1] ** 2)]
+        # パサーよりも前にいる敵ロボットIDをリストにまとめる
+        forward_their_robots_id = [robot_id for robot_id in their_robots_in_field 
+                                  if our_robots_pos[my_robot_id][0] < their_robots_pos[robot_id][0] + (abs(their_robots_vel[robot_id][0]) * dt + robot_r) * 
+                                  their_robots_vel[robot_id][0] / math.sqrt(their_robots_vel[robot_id][0] ** 2 + their_robots_vel[robot_id][1] ** 2)]
+
+        # パサーよりも前にいるロボットがいなければ空のリストを返す
+        if forward_our_robots_id == 0:
```

Comment:
forward_our_robots_idはlistにみえるので
```suggestion
        if len(forward_our_robots_id) == 0:
```